hack/deployer/runner/gcloud.go

```diff
@@ -43,6 +45,10 @@ func authToGCP(
 			return err
 		}
 
+		if gCloudProject == "" {
+			gCloudProject = "elastic-cloud-dev"
+		}
```

Comment:
I've moved this to be pulled from vault, as it's already there.

Reply:
I am confused: if we don't have `deployer-config.yaml` how can we have vault access? And why can't we have deployer config here as well? And why is duplicating the information we usually push from the buildkite pipeline now in vault an acceptable solution? And if vault is an acceptable solution why are we keeping the option to define the project via the deployer configuration. I think we should try to reduce complexity with every change instead of adding to it.

Reply:
> if we don't have deployer-config.yaml how can we have vault access?

By setting VAULT_ADDR and VAULT_ROOT_PATH. I know it's a bit confusing to have this info set via env var and the others via the deployer-config but it's not something cloud-specific, so I think it's okay. It's also aligned with https://github.com/elastic/cloud-on-k8s/issues/6561.

> why can't we have deployer config here as well?

I did not propose this because I did not find it natural but yes we could, by setting a fake `CLUSTER_NAME` or adapt set-deployer-config.sh to make it optional.
- `> E2E_PROVIDER=kind CLUSTER_NAME=n/a .buildkite/scripts/test/set-deployer-config.sh`

>  if vault is an acceptable solution why are we keeping the option to define the project via the deployer configuration.

I wasn't explicit but my idea was, if we decide to move it to Vault, to remove it from the deployer config. Otherwise it is confusing to have the two.

Reply:
Just to be clear I am fine with moving the project name to vault, but I think if we do it we should completely remove the option from the deployer config as you said it is otherwise confusing.

Reply:
I will create an issue to try and completely remove project names from code and into vault. This PR is now using deployer-config similar to other commands.